TST Ensure that sklearn/cluster/tests/test_k_means.py is seed insensitive

ShoaibKhan opened the pull request:
#### Reference Issues/PRs

Fixes sklearn/sklearn/cluster/tests/test_k_means.py in https://github.com/scikit-learn/scikit-learn/issues/22827 upstream.

#### What does this implement/fix? Explain your changes.

This upgrades the k_means tests to account for greater randomness using global_random_seed.

#### Any other comments?

Some tests were left unchanged, as suggested in [the original issue](https://github.com/scikit-learn/scikit-learn/issues/22827), as they don't test mathematical properties. Many tests failed with the introduction of randomness and to fix this, they were left as is.

tahmid-haque reviewed (changes requested):
Not sure if my branch is checkout out wrong but when running the tests a bunch failed. I ran them all at once using 
```
SKLEARN_TESTS_GLOBAL_RANDOM_SEED="all" pytest sklearn/cluster/tests/test_k_means.py -k 'test_kmeans_convergence or test_minibatch_update_consistency or test_all_init or test_minibatch_kmeans_partial_fit_init or test_fortran_aligned_data or test_minibatch_kmeans_verbose or test_kmeans_verbose or test_minibatch_sensible_reassign or test_minibatch_reassign or test_minibatch_with_many_reassignments or test_minibatch_declared_convergence or test_minibatch_iter_steps or test_kmeans_copyx or test_score_max_iter or test_predict or test_dense_sparse or test_predict_dense_sparse or test_fit_transform or test_n_init or test_float_precision or test_weighted_vs_repeated or test_scaled_weights or test_result_equal_in_diff_n_threads or test_k_means_1_iteration or test_euclidean_distance or test_inertia or test_sample_weight_unchanged or test_kmeans_plusplus_dataorder'
``` 
The command took 15 minutes to run :( Can you double check that they all pass? If not, I'd consider whether certain tests are worth converting here.

ShoaibKhan commented:
> Not sure if my branch is checkout out wrong but when running the tests a bunch failed. I ran them all at once using

Right you are, I didn't account for some tests. 

> Can you double check that they all pass? If not, I'd consider whether certain tests are worth converting here.

I'll double check all these pass and update accordingly

tahmid-haque reviewed (approved):
All good!

Tony-beeper reviewed (approved):
LGTM!

ogrisel commented:
Hello @ShoaibKhan, did you mean to do a PR to the scikit-learn main repo or to @TheBicPen's fork?

If you expect this PR to be merged to scikit-learn, you should open a new pull-request targeting the `main` branch of the main repo (https://github.com/scikit-learn/scikit-learn) instead of this fork.

ShoaibKhan commented:
> Hello @ShoaibKhan, did you mean to do a PR to the scikit-learn main repo or to @TheBicPen's fork?
> 
> If you expect this PR to be merged to scikit-learn, you should open a new pull-request targeting the `main` branch of the main repo (https://github.com/scikit-learn/scikit-learn) instead of this fork.

Hi @ogrisel, this PR was just meant for @TheBicPen's fork, but I have made another PR on the main fork.

stefmolin committed:
DOC Add note about deactivating and reactivating the conda env after installing compilers. (#24062)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jjerphan committed:
FIX Support F-contiguous arrays for `PairwiseDistancesReductions`-backed estimators (#23990)

lucyleeow committed:
DOC Update viewing docs in CI `contributing.rst` (#24040)

rprkh committed:
DOC Corrects init ndarray shape in MDS (#24067)

jjerphan committed:
MAINT Do not version *.pyc* via .gitignore (#24081)

marenwestermann committed:
FIX Convergence Warnings in Gaussian process examples (#18019)

Co-authored-by: Maren Westermann <maren.westermann@free-now.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

Vincent-Maladiere committed:
DOC Ensures that mutual_info_score passes numpydoc validation (#24091)

Vincent-Maladiere committed:
DOC Ensures that normalized_mutual_info_score passes numpydoc validation(#24093)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Maascha committed:
DOC spectral biclustering: corrected contradiction of warning and documentation for parameter method (#24098)

Vincent-Maladiere committed:
DOC Ensures that `pair_confusion_matrix` passes numpydoc validation (#24094)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lesteve committed:
FIX utils.multiclass.type_of_target with numpy 1.24 dev (#24044)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

Vincent-Maladiere committed:
DOC Ensures `v_measure_score` passes numpydoc validation (#24097)

marenwestermann committed:
TST use global_random_seed in sklearn/cluster/tests/test_bicluster.py (#23927)

Co-authored-by: Maren Westermann <maren.westermann@free-now.com>

glemaitre committed:
MAINT fix the way to call stats.mode (#23633)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Meekail Zain <34613774+Micky774@users.noreply.github.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
CI Set MACOSX_DEPLOYMENT_TARGET=10.9 (#23833)

thomasjpfan committed:
FIX Fixes OrdinalEncoder.inverse_tranform nan encoded values (#24087)

glemaitre committed:
move some issues from 1.2 to 1.1

glemaitre committed:
DOC add date for release 1.1.2

ravimakhijapi committed:
Fix Taylor expansion documentation (#24122)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

ShehanAT committed:
DOC Updates linear_model.Perceptron regarding t_ (#24073)

thomasjpfan committed:
DOC Remove deprecated multichannel parameter in example (#24116)

Micky774 committed:
MAINT Minor refactor to de-indent majority of `_assert_all_finite` (#24118)

thomasjpfan committed:
DOC Update MDS init docs in fit (#24070)

thomasjpfan committed:
MNT Use is_sparse to validate sparse data from pandas (#24074)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

ravimakhijapi committed:
DOC Fix variable name in Adaboost example (#24135)

Valentin-Laurent committed:
FIX Set n_jobs=None as default for neighbors transformers (#24075)

stefmolin committed:
DOC Update `validate_parameter_constraints()` docstring. (#24130)

thomasjpfan committed:
DOC Fixes search on webpage (#24128)

Vincent-Maladiere committed:
DOC Ensures that `rand_score` passes numpydoc validation (#24096)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

fcharras committed:
DOC Ensures that fetch_species_distributions passes numpydoc validation (#24162)

Co-authored-by: Franck Charras <franck.charras@inria.fr>

fcharras committed:
DOC Ensures that fetch_lfw_people passes numpydoc validation (#24161)

Co-authored-by: Franck Charras <franck.charras@inria.fr>

jjerphan committed:
FEA Add support for float32 on `PairwiseDistancesReduction` using Tempita (#23865)


Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

fcharras committed:
DOC Ensures that load_svmlight_files passes numpydoc validation (#24164)

Co-authored-by: Franck Charras <franck.charras@inria.fr>

Micky774 committed:
DOC Updated `adjusted_rand_score` docstring (#24019)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

thomasjpfan committed:
TST Set eigen_tol in SpectralEmbedding to stabilize test (#24125)

thomasjpfan committed:
ENH Adds text_kw to ConfusionMatrix (#24051)

jjerphan committed:
TST Fix `test_pairwise_distances_argkmin` xfail (#24168)

jjerphan committed:
MAINT Adapt `PairwiseDistancesReduction` heuristic for `strategy="auto"` (#24043)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

star1327p committed:
Moved location of legend (#24166)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

MaxwellLZH committed:
FIX Update python version in Binder environment (#24088)

Co-authored-by: Loïc Estève <loic.esteve@ymail.com>

lesteve committed:
MNT Fix binder setup (#24173)

fcharras committed:
DOC Ensures that load_svmlight_file passes numpydoc validation (#24163)

Co-authored-by: Franck Charras <franck.charras@inria.fr>

ak04p committed:
DOC Ensures that partial_dependence passes numpydoc validation (#24174)

symeneses committed:
DOC Ensures that mean_variance_axis passes numpydoc validation (#24177)

MaxwellLZH committed:
FIX adaboost return nan in feature importance (#20415)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Chiara Marmo <cmarmo@users.noreply.github.com>

lesteve committed:
MNT Handle NaNs in scipy dev rankdata (#24141)

jjerphan committed:
DOC Fix docstrings and comments for `PairwiseDistancesReduction` (#23978)



Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Meekail Zain <Micky774@users.noreply.github.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

ak04p committed:
DOC Ensures that inplace_swap_row passes numpydoc validation (#24178)

star1327p committed:
DOC Correct a typo: this -> these (#24192)

lorentzenchr committed:
DOC add missing docstring in functions in hgbt module (#24191)

fabridamicelli committed:
DOC Fix typo brier_score_loss doctring: is the sum -> as the sum (#24197)

lesteve committed:
CI Remove pandas FutureWarning in scipy-dev build (#24206)

jjerphan committed:
CI Use Ubuntu 22.04 instead of Ubuntu 18.04 (#24225)

build_tools/update_environments_and_lock_files.py has been rerun
to only generate a conda lock file for this configuration.

The previous lock file has been dropped.

stefmolin committed:
MAINT Fix section ordering and duplicate sections (#24226)

lesteve committed:
CI add ability to run float32 tests by commit message (#24202)

lucyleeow committed:
DOC Expand multilabel in decision function in glossary (#24095)

lucyleeow committed:
DOC Add section to contributing.rst on cross referencing syntax (#24092)

Vincent-Maladiere committed:
MAINT add parameters validation for ShrunkCovariance (#24108)

star1327p committed:
DOC Added links and corrected typos to plot_stock_market.py (#24209)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lucyleeow committed:
DOC Clarify LeaveOneGroupOut (#24159)

CloseChoice committed:
DOC Ensures that strip_accents_unicode passes numpydoc (#24232)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

CloseChoice committed:
DOC Ensures that non_negative_factorization passes numpydoc (#24235)

star1327p committed:
DOC Correct typos in plot_species_distribution_modeling (#24256)

lesteve committed:
DOC Remove additional semicolon in HTML for Parameters and others sections (#24252)

Vincent-Maladiere committed:
MAINT Parameters validation for `SkewedChi2Sampler` (#24114)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jjerphan committed:
DOC Add Maxwell Liu to the Contributor Experience Team (#24260)

lesteve committed:
[float32] Fix fastica tests for debian 32 bit build and global dtype float32 (#24244)

jjerphan committed:
MAINT Adapt `DistanceMetric` in prevision for fused sparse-dense support (#24223)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

stefmolin committed:
MAINT Parameter validation for `QuadraticDiscriminantAnalysis` (#24218)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

vitaliset committed:
DOC Correct typo in SelectFromModel (#24266)

vitaliset committed:
DOC Ensures that top_k_accuracy_score  passes numpydoc (#24259)

lesteve committed:
CI simplify passing of environment variables in Azure docker template (#24251)

Micky774 committed:
MAINT Update the name scheme of the `PairwiseDistancesReduction` backend (#24077)

kasmith11 committed:
MAINT add parameters validation for SplineTransformer (#24057)

thomasjpfan committed:
CI Update to macOS-11 in Azure CI (#24267)

cmarmo committed:
CI Update to MacOS-11 in azure build for MacOS-10.15 is deprecated. (#24291)

NimaSarajpoor committed:
DOC Fix minor typo in clustering user guide (#24284)

stefmolin committed:
MAINT Parameter validation for `TransformedTargetRegressor` (#24219)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

stefmolin committed:
MAINT Add parameter validation to `PolynomialFeatures` (#24214)

stefmolin committed:
MAINT Add parameter validation to SequentialFeatureSelector. (#24129)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

IvanLauLinTiong committed:
DOC Ensure dict_learning_online passes numpydoc validation (#24289)

star1327p committed:
DOC added description for DBSCAN (#24278)

yuanx749 committed:
DOC Fix some typos (#24300)

thomasjpfan committed:
CI Fixes junit in CI based on docker (#24299)

stefmolin committed:
MAINT Add parameter validation to RFE and RFECV. (#24137)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
MNT Adds dict typing hint to _parameter_constraints for mypy (#24301)

mehmetcanakbay committed:
MAINT _validate_params in BayesianRidge (#24047)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

shellyfung committed:
MAINT Parameter validation for LedoitWolf (#24132)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Vincent-Maladiere committed:
MAINT Parameters validation for `SimpleImputer` (#24109)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

ArturoAmorQ committed:
DOC Rework plot_adjusted_for_chance_measures.py example (#23708)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

rprkh committed:
MAINT Parameter Validation for Lars, LarsCV, LassoLars, LassoLarsCV and LassoLarsIC (#24033)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

jeremiedbb committed:
ENH Param validation: Introduce Options constraint (#24048)

stefmolin committed:
MAINT Add parameter validation for classes in _univariate_selection.py (#24139)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

amitalokbera committed:
MAINT Parameters validation for AdditiveChi2Sampler (#24149)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

stefmolin committed:
MAINT Add parameter validation to `FeatureHasher` (#24179)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

tbonald committed:
Add scikit-network to related projects (#24304)

stefmolin committed:
MAINT Add parameter validation for `FunctionTransformer`. (#24180)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

stefmolin committed:
MAINT Parameter validation for `PolynomialCountSketch` (#24217)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

stefmolin committed:
MAINT Parameter validation for `RBFSampler` (#24220)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

shellyfung committed:
MAINT Parameter validation for Isomap (#24183)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

stefmolin committed:
MAINT Add parameter validation for `LabelPropagation` and `LabelSpreading` (#24211)

stefmolin committed:
MAINT Add parameter validation to `IterativeImputer` (#24210)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

olukowski committed:
MAINT Parameters validation for `CalibratedClassifierCV` (#24101)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

stefmolin committed:
MAINT Add parameter validation to `LatentDirichletAllocation` (#24212)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

kasmith11 committed:
MAINT Parameters validation for `SpectralEmbedding` (#24103)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

olukowski committed:
MAINT Parameters validation for `ClassifierChain` and `RegressorChain` (#24112)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

laurburke committed:
DOC Updating Branding Guidelines (#24241)

stellalin7 committed:
MAINT Parameter validation for SpectralBiclustering and SpectralCoclustering (#24170)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

star1327p committed:
DOC improved readability of instructions (#24311)

naoise-h committed:
MAINT Param validation: remove n_estimators checks from validate_estimator (#24224)

Co-authored-by: Meekail Zain <34613774+Micky774@users.noreply.github.com>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

EdAbati committed:
MAINT Parameters validation for NeighborhoodComponentsAnalysis (#24195)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

stefmolin committed:
MAINT Add parameter validation to `HashingVectorizer`. (#24181)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

stefmolin committed:
MAINT Add parameter validation to `SelectFromModel` (#24213)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

stefmolin committed:
MAINT Parameter validation for `PatchExtractor` (#24215)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

OmarManzoor committed:
MAINT Parameters validation for MultiTaskElasticNet and MultiTaskLasso (#24295)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

svenstehle committed:
MNT add dtype preservation tag to BernoulliRBM (#24318)

pmli committed:
MAINT lars_path_gram passes numpydoc validation (#24319)

Mehgarg committed:
MAINT Parameters validation for RidgeCV and RidgeClassifierCV (#24184)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

OmarManzoor committed:
MAINT Parameters validation for OneVsOneClassifier (#24292)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

MisaOgura committed:
MAINT Validate params in OAS (#24324)

ShisuiUzumaki committed:
MAINT Parameters validation for DictionaryLearning (#24242)

Co-authored-by: Meekail Zain <zainmeekail@gmail.com>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

hristo-vrigazov committed:
MAINT coverage_error passes numpydoc validation (#24322)

betatim committed:
DOC Rework `plot_kmeans_stability_low_dim_dense.py` to make it more colorblind friendly (#24323)

thomasjpfan committed:
FIX Allow for KMean's attributes to be readonly (#24258)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

nitinramvelraj committed:
MAINT Parameters validation for OneVsRest estimator  (#24290)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

maikia committed:
DOC Ensure sklearn.linear_model._omp.orthogonal_mp_gram passes numpydoc validation (#24329)

maikia committed:
DOC Ensure manifold._locally_linear.locally_linear_embedding passes numpydoc validation (#24330)

maikia committed:
DOC Ensure manifold._t_sne.trustworthiness passes numpydoc validation (#24333)

mrastgoo committed:
MAINT Parameters validation for Nystroem (#24332)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

OmarManzoor committed:
MAINT Parameters validation for MiniBatchDictionaryLearning (#24271)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

ichko committed:
DOC Ensures that mean_pinball_loss passes numpydoc validation (#24336)

mikeonly committed:
MAINT Parameters validation for RANSACRegressor (#24339)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

OmarManzoor committed:
MAINT Parameters validation for OPTICS (#24326)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

star1327p committed:
DOC moved legend to the right (#24342)

Safikh committed:
DOC Ensures that adjusted_rand_score passes numpydoc validation (#24345)

Safikh committed:
DOC Ensures that consensus_score passes numpydoc validation (#24343)

maikia committed:
DOC Ensure plot_det_curve passes numpydoc validation (#24334)

maikia committed:
DOC Ensure plot_partial_dependence passes numpydoc validation (#24325)

maikia committed:
DOC Ensure dict_learning passes numpydoc validation (#24316)

maikia committed:
DOC Ensure externals._packaging.version.parse passes numpydoc validation (#24320)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Safikh committed:
DOC Ensures that adjusted_mutual_info_score passes numpydoc validation (#24344)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

TheDevPanda committed:
DOC ensure sklearn/utils/extmath/stable_cumsum passes numpydoc (#24348)

jjerphan committed:
MAINT Use the newest NumPy C API where possible (#24049)

MarcoGorelli committed:
MAINT clean-up unused variables and imports in cython files (#24347)

OmarManzoor committed:
DOC Ensures that `scale` passes numpydoc validation (#24362)

OmarManzoor committed:
DOC Ensures that `squared_norm` passes numpydoc validation (#24360)

OmarManzoor committed:
DOC Ensures that `maxabs_scale` passes numpydoc validation (#24359)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

3lLobo committed:
DOC fix a typo in `build_analyzer` docstring (#24309)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Meekail Zain <34613774+Micky774@users.noreply.github.com>

shellyfung committed:
DOC Ensures that sklearn.metrics._ranking passes numpydoc validation (#24351)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

shellyfung committed:
DOC Ensures that cluster._supervised passes numpydoc validation (#24352)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

gittasche committed:
MAINT abstracmethod in `_gb_losses` (#24277)

lesteve committed:
FIX np.divide undefined behaviour with where in gaussian processes (#24245)

ogrisel committed:
MAINT update lock files on CI (#24363)

ShehanAT committed:
FIX Enables label_ranking_average_precision_score to support sparse y_true (#23442)

Allow y_true to be in CSR format.

star1327p committed:
DOC revised description for Pipeline ANOVA SVM (#24358)

Vincent-Maladiere committed:
MAINT Remove `x_squared_norms` arg from `k_means_lloyd` signature (#24264)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

lucyleeow committed:
amend wording doc ci (#24370)

lucyleeow committed:
rm custom_references_resolver (#24366)

biswaroopmitra committed:
DOC Fix grammar in HashingVectorizer's docstring (#24372)

lesteve committed:
CI Pin numpy for conda defaults channel (#24373)

lesteve committed:
CI Update nogil lock files (#24377)

kshitijkapadni committed:
DOC Ensures that l1_min_c passes numpydoc validation (#24134)

thomasjpfan committed:
MNT Update doc and param validation for SequentialFeatureSelector about scoring (#24296)

thomasjpfan committed:
CI Cancel in progress wheel builds (#24375)

zadilkhwaja committed:
DOC ensures sklearn.utils.shuffle passes numpydoc validation (#24367)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lesteve committed:
FIX remove np.divide with where and without out argument in `precision_recall_curve` (#24382)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lesteve committed:
CI: pin pypy to avoid cloudpickle issue (#24376)

jeremiedbb committed:
TST Remove remaining tests now covered by common param validation test (#24383)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: criterion="mse"/"mae" in trees (#24384)

lucyleeow committed:
DOC Expand types of docs in contributing docs section (#24371)

jeremiedbb committed:
cln deprecations (#24385)

jeremiedbb committed:
cln deprecations (#24386)

elcorto committed:
DOC: Streamline usage of Gaussian Process reference book (#24331)

Co-authored-by: Loïc Estève <loic.esteve@ymail.com>

lucyleeow committed:
DOC Add note on backtick rendering (#24368)

star1327p committed:
DOC moved legend to center (#24379)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

jeremiedbb committed:
MAINT Clean deprecation for 1.2: n_features_ replaced by n_featues_in_ (#24388)

jjerphan committed:
MAINT Pop `Y_norm_squared` from `metric_kwargs` when needed (#24272)

thomasjpfan committed:
CI Race condition work around for doc-min-dependencies (#24400)

betatim committed:
DOC Use pipeline in classifiers example (#24399)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: BIRCH attributes (#24398)

glemaitre committed:
MAINT Clean deprecation for 1.2: learning_rate and init in TSNE (#24389)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

EdAbati committed:
MAINT rename and deprecate `base_estimator` in favor of `estimator` in ensemble classes (#23819)

Co-authored-by: Adrian Trujillo Duron <adrian.td96@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

rprkh committed:
ENH Add dtype preservation to SkewedChi2Sampler (#24350)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

glemaitre committed:
MAINT black compliant

Kshitij68 committed:
DOC Ensures plot_precision_recall_curve passes the numpydoc validation (#24403)

betatim committed:
ENH Preserves dtype in `RBFSampler` (#24317)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jeremiedbb committed:
MAINT Fix some broken asv benchmarks (#24407)

thomasjpfan committed:
ENH Always set input feature attributes in FunctionTransformer (#23993)

glemaitre committed:
TST refactor test for contamination param for outlier detectors (#23738)

lucyleeow committed:
DOC Clarify when GroupKFold same as LeaveOneGroupOut (#24104)

x110 committed:
DOC remove Matplotlib warning in plot_varimax_fa example (#24397)

Co-authored-by: Loïc Estève <loic.esteve@ymail.com>

rprkh committed:
DOC RandomizedSearchCV verbosity parameter description (#24413)

zadilkhwaja committed:
DOC ensures sklearn.utils.safe_mask passes numpydoc validation (#24425)

christianwaldmann committed:
MNT Update license year to 2022 (#24418)

biswaroopmitra committed:
DOC improve and correct typo in docstring (#24417)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

rhelmeczi committed:
DOC fix some typos in docstring (#24416)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: Ransac losses (#24408)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: KernelPCA (#24406)

OmarManzoor committed:
FIX Ensure that GaussianProcessRegressor predict method does not modify input (#24405)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: NMF (#24393)

Kshitij68 committed:
API Deprecate `n_feature_` fitted attribute in PCA (#24421)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: losses in gradient boosting (#24394)

francesco-tuveri committed:
DOC Rework `plot_affinity_propagation.py` to make it colorblind friendly (#24335)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

ArturoAmorQ committed:
FEA Add support for micro averaging in ovr-roc-auc (#24338)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

ArturoAmorQ committed:
DOC Rework plot_grid_search_text_feature_extraction.py example (#23740)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

zadilkhwaja committed:
DOC ensures sklearn.utils.safe_sqr passes numpydoc validation (#24437)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: assert_warns (#24442)

betatim committed:
ENH Remove validation from `__init__` for `SGDOneClassSVM` (#24433)



Co-authored-by: iofall <50991099+iofall@users.noreply.github.com>
Co-authored-by: arisayosh <15692997+arisayosh@users.noreply.github.com>

michaelmccutchen committed:
DOC Ensures that sparsefuncs.count_nonzero passes numpydoc validation (#24447)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

thomasjpfan committed:
FEA Adds class_weight to HistGradientBoostingClassifier (#22014)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

Vincent-Maladiere committed:
EHN Add multilabel classification support for `StackingClassifier` (#24146)

* Add stacking multilabel functionality

* Add underscore to a class attr

* Remove model from base estimator in test_stacking

* Remove scale in train/test split in test_stacking_classifier_multilabel

* Add stack_method as a test parameter, change RandomForestClassifier to KNeighborsClassifier in test

* Update Changelog

* fix doc typos

* predict_proba output will be concatenate this list in an array of shape n_samples, n_outputs * n_classes - 1. Update test.

* Update sklearn/ensemble/_stacking.py

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

* Update doc/whats_new/v1.0.rst

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

* update whats_new

* add passthrough test

* update whats_new with current PR

* Apply suggestions from code review

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

* update tests

* Apply suggestion to update comments on `concatenate`

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

* parametrized the two tests into one

* parametrized the two tests into one

* strip the mysterious trailing  _r

* fix multilabel list scenario

* add Guillaume's recommendations

* add test for

* some fix

* split tests

* fix flake8

* add suggestions

* Trigger CI

* remove multiclass-multioutput from comments and docstrings

Co-authored-by: Nicolas <nicolas.peretti@fligoo.com>
Co-authored-by: Nestor Navarro <nestornav@gmail.com>
Co-authored-by: Nestor Navarro <nestor.navarro@mercadolibre.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

betatim committed:
DOC Update estimator representation example (#24439)

rusdes committed:
API Rename OneHotEncoder option sparse to sparse_output (#24412)

Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

zadilkhwaja committed:
DOC ensures sklearn.utils.multiclass.class_distribution passes numpydoc validation (#24452)

jjerphan committed:
DOC Fix `monotonic_cst` description in HGBT docstrings (#24396)



Co-authored-by: "Samuel O. Ronsin" <samuel.ronsin@dataiku.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>

psteinb committed:
DOC fix labels in legend of plot_logistic example (#24455)

yuanx749 committed:
DOC Ensures that check_is_fitted passes numpydoc validation (#24454)

jeremiedbb committed:
MAINT Clean deprecation for 1.2: plot functions (#24445)

ogrisel committed:
MAINT Update CI dependencies (#24453)

* MAINT Update CI dependencies

* Trigger [pypy] [scipy-dev] [nogil] [icc-build]

michaelmccutchen committed:
DOC ensures sklearn.utils.sparsefuncs.csc_median_axis_0 passes numpydoc validation (#24461)

lucyleeow committed:
fix link (#24468)

lucyleeow committed:
DOC Improve format in Documentation section of contributing.rst (#24467)

zadilkhwaja committed:
passes validation (#24463)

jjerphan committed:
CI Make pytest err on PytestUnraisableExceptionWarning (#24438)

Co-authored-by: Jérémie du Boisberranger <jeremiedbb@users.noreply.github.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

rusdes committed:
Fixed AgglomerativeClustering's distance_threshold misleading statement (#24460)

Co-authored-by: joaocmd <joaodavid1999@hotmail.com >

lucyleeow committed:
DOC Improve format in code examples of splitters (#24466)

jjerphan committed:
FEA Fused sparse-dense support for `PairwiseDistancesReduction`  (#23585)



Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>
Co-authored-by: Jérémie du Boisberranger <jeremiedbb@users.noreply.github.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Meekail Zain <Micky774@users.noreply.github.com>

yuanx749 committed:
DOC Ensures that incr_mean_variance_axis passes numpydoc validation (#24477)

zadilkhwaja committed:
DOC ensures sklearn.utils.multiclass.unique_labels passes numpydoc validation. (#24476)

klieret committed:
[Doc] Fix hyperlink formatting (#24457)

* [Doc] Fix hyperlink formatting

* Use doi for reference

jeremiedbb committed:
MAINT Clean deprecation for 1.2: DictionaryLearning (#24471)

jeremiedbb committed:
cln deprecations (#24478)

RenTrieu committed:
DOC: Added auto solver table under Ridge Regression in Linear Models (#23297)



Co-authored-by: recovery1 <recovery1@Leeks>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Meekail Zain <34613774+Micky774@users.noreply.github.com>

davidgilbertson committed:
DOC Update loss from LAD to absolute_error (#24485)

thomasjpfan committed:
ENH Adds Array API support to LinearDiscriminantAnalysis (#22554)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

leonardbinet committed:
FIX properly support sparse matrices in type_of_target (#14862)

Co-authored-by: Leonard Binet <lbinet@alkemics.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Bartosz Telenczuk <bartosz.telenczuk@gmail.com>
Co-authored-by: Ilia Ivanov <ilia.ivanov@olx.com>
Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

jeremiedbb committed:
MAINT Clean deprecation for 1.2: sigma_ in GaussianNB (#24492)

* cln deprecations

* lint

jeremiedbb committed:
MAINT Clean deprecation for 1.2: validation (#24493)

* cln deprecations

* cln

* fix tst switch to pytest.raises

marenwestermann committed:
TST use global_random_seed in sklearn/cluster/tests/test_affinity_propagation.py (#23838)

Harsh14901 committed:
Fix Erroneous Avg. Loss calculation in sgd (#23798)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>

jeremiedbb committed:
Comment uses of deprecated plot functions in past release highlights (#24456)

jeremiedbb committed:
TST Speed up tests for pairwise distances reductions (#24479)

jeremiedbb committed:
cln deprecations (#24480)

cmarmo committed:
FIX Fix ranking for scipy >= 1.10. (#24483)


Co-authored-by: Meekail Zain <34613774+Micky774@users.noreply.github.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

diederikwp committed:
ENH Implement get_feature_names_out for FeatureUnion in case of "passthrough" (#24058)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

seljaks committed:
DOC ensures that two sklearn.utils.graph funcs pass numpydoc (#24474)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jeremiedbb committed:
adjust global_random_seed enabled tests

jeremiedbb committed:
Merge remote-tracking branch 'upstream/main' into pr/ShoaibKhan/23388

jeremiedbb committed:
lint

jeremiedbb committed:
test [all random seeds]
test_kmeans_elkan_results
test_kmeans_convergence
test_minibatch_update_consistency
test_fortran_aligned_data
test_minibatch_sensible_reassign
test_minibatch_reassign
test_score_max_iter
test_dense_sparse
test_transform
test_fit_transform
test_n_init
test_k_means_function
test_float_precision
test_weighted_vs_repeated
test_unit_weights_vs_no_weights
test_scaled_weights
test_result_equal_in_diff_n_threads
test_k_means_1_iteration
test_euclidean_distance
test_inertia
test_kmeans_plusplus_output
test_kmeans_plusplus_dataorder
